### PR TITLE
Bug 1294478 - Move JSON serialization out of the templates and into perl to improve performance

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
[Bug 1294478 - Move JSON serialization out of the templates and into perl to improve performance](https://bugzilla.mozilla.org/show_bug.cgi?id=1294478)

### CODE REVIEW

File: enter_bug.cgi [hunk 1]
```diff
@@ -438,6 +439,32 @@ my $format = $template->get_format("bug/create/create",
...
+    foreach my $f (@{ $c->flag_types->{attachment} }) {
+        push (@flag_list, $f->id);
+    }
+    $data{flags}[ $count ] = join(',', @flag_list);
+    $count ++;
+    if ( Bugzilla->params->{'useqacontact'} ) {
+        $data{initialqacontacts}[ $count ] = $c->default_qa_contact->login
+    }
+    my @initial_cc_list;
+    foreach my $cc_user (@{ $c->flag_types->{bug} }) {
```
Comment (dylanwh) [inline enter_bug.cgi]: I think you meant to say @{ $c->initial_cc }, not $c->flag_types. I corrected for that and the code seems
to work on first pass.

File: template/en/default/bug/create/create.html.tmpl [hunk 2]
```diff
@@ -63,43 +63,17 @@ function initCrashSignatureField() {
                       'cf_crash_signature_action', 'cf_crash_signature');
   [% END %]
 }
-
-var initialowners = new Array([% product.components.size %]);
+var data = [% data %];
```
Comment (dylanwh) [inline template/en/default/bug/create/create.html.tmpl]: Even though we think data is json, it's important to make the browser parse it so arbitrary JS can never leak in.
var data = JSON.parse('[% data FILTER js %]');

File: template/en/default/bug/create/create.html.tmpl [hunk 3]
```diff
@@ -63,43 +63,17 @@ function initCrashSignatureField() {
...
-var components = new Array([% product.components.size %]);
-var comp_desc = new Array([% product.components.size %]);
-var flags = new Array([% product.components.size %]);
+var initialccs = data["initialccs"]
+var components = data["components"];
+var comp_desc = data["comp_desc"];
+var flags = data["flags"];
 [% IF Param("useqacontact") %]
-    var initialqacontacts = new Array([% product.components.size %]);
+    var initialqacontacts = data["initialqacontacts"]
```
Comment (dylanwh) [inline template/en/default/bug/create/create.html.tmpl]: nit: missing semi-colon.

File: template/en/default/bug/create/create.html.tmpl [hunk 4]
```diff
@@ -63,43 +63,17 @@ function initCrashSignatureField() {
...
-
-var initialowners = new Array([% product.components.size %]);
+var data = [% data %];
+var initialowners = data["initialowners"];
 var last_initialowner;
-var initialccs = new Array([% product.components.size %]);
-var components = new Array([% product.components.size %]);
-var comp_desc = new Array([% product.components.size %]);
-var flags = new Array([% product.components.size %]);
+var initialccs = data["initialccs"]
```
Comment (dylanwh) [inline template/en/default/bug/create/create.html.tmpl]: nit: missing semi-colon.

**Comment by martiansideofthemoon** [comment]
@dylanwh does this look better?

**Comment by dylanwh** [comment]
I moved this to a new bug so we can review and land it. Can you update the commit message? @martiansideofthemoon

**Comment by martiansideofthemoon** [comment]
@dylanwh r?

**Comment by floatingatoll** [comment]
@dylanwh, this seems reasonable to me. Does it apply cleanly still?

**Comment by dylanwh** [comment]
it's not in a working state, no. Probably best to just try again later.

**Pull request closed without merging**